modules/apps/list-type/list-type-test/src/testIntegration/java/com/liferay/list/type/internal/upgrade/v1_3_2/test/ListTypeDefinitionUpgradeProcessTest.java

```diff
@@ -0,0 +1,118 @@
+import com.liferay.list.type.model.ListTypeDefinition;
+import com.liferay.list.type.service.ListTypeDefinitionLocalService;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.Collections;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Igor Costa
+ */
+@RunWith(Arquillian.class)
+public class ListTypeDefinitionUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_user = UserTestUtil.addUser(TestPropsValues.getCompanyId());
+
+		_listTypeDefinition =
+			_listTypeDefinitionLocalService.addListTypeDefinition(
+				null, _user.getUserId(),
+				Collections.singletonMap(
+					LocaleUtil.US, RandomTestUtil.randomString()),
+				false,
+				Collections.singletonList(
+					ListTypeEntryUtil.createListTypeEntry(
+						RandomTestUtil.randomString())));
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		_listTypeDefinitionLocalService.deleteListTypeDefinition(
+			_listTypeDefinition);
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator,
+			"com.liferay.list.type.internal.upgrade.v1_3_2." +
+				"ListTypeDefinitionUpgradeProcess");
+
+		long userId = _user.getUserId();
+
+		_userLocalService.deleteUser(_user);
+
+		Assert.assertTrue(_userLocalService.fetchUser(userId) == null);
+
+		Assert.assertTrue(_listTypeDefinition.getUserId() == userId);
+
+		upgradeProcess.upgrade();
+
+		User defaultServiceAccountUser =
+			_userLocalService.fetchUserByScreenName(
+				TestPropsValues.getCompanyId(), "default-service-account");
+
+		_listTypeDefinition =
+			_listTypeDefinitionLocalService.
+				getListTypeDefinitionByUuidAndCompanyId(
+					_listTypeDefinition.getUuid(),
+					TestPropsValues.getCompanyId());
+
+		Assert.assertTrue(
+			_listTypeDefinition.getUserId() ==
+				defaultServiceAccountUser.getUserId());
```

Comment:
I would organize this method like this:
```
long userId = _user.getUserId();

_userLocalService.deleteUser(_user);

//you need to refactor
Assert.assertTrue(_userLocalService.fetchUser(userId) == null);

//you need to refactor
Assert.assertTrue(_listTypeDefinition.getUserId() == userId);

UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
	_upgradeStepRegistrator,
	"com.liferay.list.type.internal.upgrade.v1_3_2." +
		"ListTypeDefinitionUpgradeProcess");

upgradeProcess.upgrade();

User defaultServiceAccountUser =
	_userLocalService.fetchUserByScreenName(
		TestPropsValues.getCompanyId(), "default-service-account");

_listTypeDefinition =
	_listTypeDefinitionLocalService.
		getListTypeDefinitionByUuidAndCompanyId(
			_listTypeDefinition.getUuid(),
			TestPropsValues.getCompanyId());

//you need to refactor
Assert.assertTrue(
	_listTypeDefinition.getUserId() ==
		defaultServiceAccountUser.getUserId());
```

Reply:
It is better to keep `UpgradeProcess upgradeProcess = ...` close to its usage `upgradeProcess.upgrade();`